### DISCUSSION TITLE
Build Umgebung für freetz-ng

### DISCUSSION THREAD

**Discussion by janfreetz** [Freetz-NG/freetz-ng#159] [Allgemein]
Hallo Developer,
wo finde ich ein Virtualbox VDI Datei für die Standard Build Umgebung?
Oder wie erstelle ich mir eine default Build Umgebung, damit ich / wir auf gleichem Stand sind?
Die soll für 154.07.21 rev82154 {ALL} [PSQ19] sein.
https://osp.avm.de/fritzbox/fritzbox-7590/
source-files-FRITZ.Box_7590-07.21.tar.gz   
VG
Jan

**Comment by fda77** [comment]
Hallo Benutzer, jedes halbwegs aktuelle Linux sollte passen. Mindestens diese Pakete werden benötigt: https://freetz-ng.github.io/freetz-ng/wiki/10_Beginner/install.de.html#installation-der-ben%C3%B6tigten-pakete-ubuntu
Oder wie viele eine fertige VM von Gismotro benutzen, siehe readme unten auf https://github.com/Freetz-NG/freetz-ng

**Comment by fda77** [comment]
Es gibt übrigens oben das "Discussions", "Issues" sind für Fehler

**Comment by janfreetz** [comment]
Ok, sorry. 
Wo sollte man allgemeine Fragen stellen?

https://github.com/Freetz-NG/freetz-ng/discussions

Link zuvor?

**Comment by fda77** [comment]
Wie gesagt, zu "Discussions", das ist eine neue Funktion von Github.
Q&A steht für Fragen&Antworten

**Comment by janfreetz** [comment]
Viel Erfolg beim Neubeginn.

Q&A steht für Fragen&Antworten

Was willst Du damit sagen, Q&A finde ich nirgends.
Q&A = Discussions?

**Comment by fda77** [comment]
Das ist ne Kategorie in den Diskussionen

**Comment by PeterPawn** [comment]
Wenn das Problem (https://github.com/github/feedback/discussions/2952) gefixt wurde, kann man auch wieder Issues in Discussions umwandeln - das hält die "Fehlerliste" klein und übersichtlich (was auch bei "closed" das Ziel sein sollte, damit andere bei der Suche nicht über zu viele falsch zugeordnete Themen iterieren müssen).

**Comment by fda77** [comment]
Prima dass das bald kommt. Ich hatte das hinter "Transfer" erwartet, aber das wird man wohl nicht so oft (nie) brauchen

**Comment by fda77** [comment]
converted @PeterPawn

**Comment by Gismotro** [comment]
Tach zusammen,
@fda77 : Ich baue keine VM mehr, aber die PDF unter dem Link wird immer auf aktuellem Stand gehalten genauso wie das HowTo im IPPF: https://www.ip-phone-forum.de/threads/freetz-ng-fehler-beim-bauen-f%C3%BCr-unterschiedliche-konfigurationen.308030/#post-2387896

**Reply by fda77** [reply]
Warum hast du die VMs denn gleich gelöscht? Die LTS hättest du doch stehen lassen können. Der Vorgänger von cinereious ist heut noch verfügbar.
Die Anleitung find ich suboptimal:
 - warum soll man das Runelvel auf 3 ändern? Nimm die Minimal/Serverversion, die kein X
 - warum soll man die Sprache ändern? Das kann man beim installieren auswählen
 - warum soll man ssh nachinstallieren? Das kann man beim installieren auswählen
 - umask würd ich einfach in .profile ändern
 - Die Liste der prerequisites ändert sich öfter mal, wie du gemerkt hast. Verlink einfach die Quelle

 Bleibt eigentlich nur noch ftp & samba zu konfigurieren (kann man bestimmt beim installieren auch auswählen)

**Reply by Gismotro** [reply]
OK, VM (Freetz-Linux-1.5.8-Ubuntu 20.04.01 LTS (Focal Fossa) 64-Bit.ova) ist wieder verfügbar, aber wir müssen halt drauf achten das die Nutzer die das nutzen wollen halt dran denken die VM nach deinen Vorgaben aktualisieren müssen.
Cool wäre der Hinweis gleich hier mit zu verankern : Requirements: https://github.com/Freetz-NG/freetz-ng/blob/master/README.md nach dem Motto : wenn VM von Gismotro, dann mache auch Schriit Ubuntu 20 64-Bit: https://github.com/Freetz-NG/freetz-ng/blob/master/docs/PREREQUISITES.md

Dann braucht man nur eine Stelle zu aktualiesieren und die Foren können dahin verlinken.

**Reply by fda77** [reply]
"Euer Freetz" und "deine Vorgaben"?
Wenn irgendwas erforderliches nicht gefunden wird kommt doch schon seit langem diese Meldung: https://github.com/Freetz-NG/freetz-ng/blob/master/Makefile#L135
```
Some build prerequisites are missing! Please install the missing packages before trying again.
See https://freetz-ng.github.io/freetz-ng/PREREQUISITES for installation hints
```
Ich wüsste jetzt nicht wie man das noch optimieren könnte. Die URL muss halt jeder selbst öffnen und apt-get ausführen.
Solang man nicht die user-competence "aptguru" (https://github.com/Freetz-NG/freetz-ng/blob/master/config/ui/user-competence.in#L35) ausgewählt hat sind es nur die nötigsten die AVM neuerdings braucht. In dem Fall muss man halt viel Zeit zum bauen mitbringen

**Reply by Gismotro** [reply]
Hast Recht. Schauen wir mal ob es klappt.

**Comment by Gismotro** [comment]
Neues Update liegt auf dem Teamserver: "Freetz-Linux-1.7.1-Ubuntu 21.04-(Hirsute Hippo) 64-Bit"
